src/Proxy.hs

```diff
@@ -1083,6 +1085,20 @@ putTransferR =
                     Left err -> fail err
                     Right tx -> return tx
 
+putTransferBytesR :: Handler TypedContent
+putTransferBytesR = do
+    binData <- runConduit $ rawRequestBody .| sinkLbs
+    case S.decodeLazy binData of
```

Comment:
I would suggest using `runGetLazy (getBareBlockItem SP8) binData`, and deserialize a block item rather than just a normal transaction.